pkg/store/store.go

```diff
@@ -187,28 +189,54 @@ func NewCacheStoresFromObjs(objs ...runtime.Object) (CacheStores, error) {
 // The CacheStore must be initialized (see NewCacheStores()) or this will panic.
 func (c CacheStores) Add(obj runtime.Object) error {
 	switch obj := obj.(type) {
+	// ----------------------------------------------------------------------------
+	// Kubernetes Core API Support
+	// ----------------------------------------------------------------------------
 	case *extensions.Ingress:
 		return c.IngressV1beta1.Add(obj)
 	case *networkingv1.Ingress:
 		return c.IngressV1.Add(obj)
-	case *configurationv1beta1.TCPIngress:
-		return c.TCPIngress.Add(obj)
-	case *configurationv1alpha1.UDPIngress:
-		return c.UDPIngress.Add(obj)
 	case *corev1.Service:
 		return c.Service.Add(obj)
 	case *corev1.Secret:
 		return c.Secret.Add(obj)
 	case *corev1.Endpoints:
 		return c.Endpoint.Add(obj)
-	case *configurationv1.KongPlugin:
+	// ----------------------------------------------------------------------------
+	// Kong API Support
+	// ----------------------------------------------------------------------------
+	case *kongv1.KongPlugin:
+		return c.Plugin.Add(obj)
+	case *kongv1.KongClusterPlugin:
+		return c.ClusterPlugin.Add(obj)
+	case *kongv1.KongConsumer:
+		return c.Consumer.Add(obj)
+	case *kongv1beta1.TCPIngress:
+		return c.TCPIngress.Add(obj)
+	case *kongv1alpha1.UDPIngress:
+		return c.UDPIngress.Add(obj)
+	// ----------------------------------------------------------------------------
+	// Legacy Kong API Support
+	// ***********************
+	//
+	// TODO: we expect to be able to remove these older duplicate APIs in favor of
+	//       the ones above prior to releasing KIC 2.0.
+	//       SEE: https://github.com/Kong/kubernetes-ingress-controller/milestone/12
```

Comment:
Could you please create a specific issue for removal of legacy APIs from Store and implementing a controller that autonegotiates the api version?

Reply:
:+1:

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/1217

```suggestion
	//       SEE: https://github.com/Kong/kubernetes-ingress-controller/issues/1217
```